src/presentation/controller/registration/RegistrationController.ts

```diff
@@ -0,0 +1,103 @@
+import { NextFunction, Request, RequestHandler, Response } from "express";
+
+import RegistrationService from "../../../application/registration/Service";
+import PageRegistrationType from "./PageRegistrationType";
+import registrationsRouting, { NEXT2_URL } from "./Routing";
+import { NameEndingType } from "@companieshouse/api-sdk-node/dist/services/limited-partnerships";
+import Controller from "../Controller";
+import TransactionLimitedPartnership from "domain/entities/TransactionLimitedPartnership";
+import { PageRouting } from "../PageRouting";
+
+class RegistrationController extends Controller {
+  private registrationService: RegistrationService;
+
+  constructor(registrationService: RegistrationService) {
+    super();
+    this.registrationService = registrationService;
+  }
+
+  getPageRouting(): RequestHandler {
+    return (request: Request, response: Response, next: NextFunction) => {
+      try {
+        const pageType = super.pageType(request.path);
+
+        const result = super.getRouting(
+          registrationsRouting,
+          pageType as PageRegistrationType
+        );
+
+        const pageRouting = super.inserIdsInAllUrl(
```

Comment:
As discussed, this could be done within the call to `super.getRouting` rather than here, so that the routing object you get back is always 'valid', i.e. has the ids already populated